plop-templates/tsconfig.json.hbs

```diff
@@ -0,0 +1,22 @@
+// Generated by seam-plop
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "strict": false,
```

Comment:
looked into this a bunch today in the context of converting some projects over. I think strict mode should be a requirement for libraries and recommended for APIs, but not applications (especially frontend applications). We should always default to `"strict": true` A bit on my process:

I converted Seam for Homes to strict mode and looked at all the errors, I think the typing would fix a lot of errors, but it was also clear that a lot of functions would have pretty large signatures, and some of the errors that came up only in strict mode were difficult to debug because of complex type involvement. There were also a lot of errors (195), I found some examples where I thought the addition of types was good/bad (some examples below but not showing any particular point)

![image](https://user-images.githubusercontent.com/1910070/181170368-d016fc23-6145-4243-8032-c34623fc7b4e.png)

![image](https://user-images.githubusercontent.com/1910070/181169579-b95efa20-477b-4be3-afc7-5f61aaa8dab0.png)

I want to be careful to make recommendations that don't impact velocity, I think strict mode would have helped with a lot of bugs on seam for homes, and ultimately it's not preventing the use of `any` to quickly get around issues and write bad code, but it makes it more clear where the programmer is taking a shortcut.

TLDR; strict mode is probably good